fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/TypeRdfContext.java

```diff
@@ -51,15 +55,15 @@ public TypeRdfContext(final FedoraResource resource,
             throws RepositoryException {
         super(resource, idTranslator);
 
-        //include rdf:type for primaryType, mixins, and their supertypes
-        concat(resource().getTypes().stream()
-                .map(uri -> create(subject(), type.asNode(), createURI(uri.toString())))
-                .iterator());
-        if (resource() instanceof NonRdfSource) {
+        final Stream.Builder<Triple> other = Stream.builder();
+        if (resource instanceof NonRdfSource) {
             // gather versionability info from the parent
-            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
-                concat(create(subject(), type.asNode(), createURI(MIX_NAMESPACE + "versionable")));
+            if (resource.getNode().getParent().isNodeType("mix:versionable")) {
+                other.accept(create(subject(), type.asNode(), createURI(MIX_NAMESPACE + "versionable")));
             }
         }
+        this.stream = Stream.concat(
```

Comment:
You are changing the `*RdfContext` classes to set `this.stream` directly, vs. `concat()`ing to whatever used to be in `this.stream`. This may result in the unexpected loss of triples in `*RdfContext` classes in the inheritance tree above or below any of your updated `*RdfContext` classes.